pkg/binding/controller.go

```diff
@@ -99,17 +99,27 @@ type Controller struct {
 func NewController(parentLogger logr.Logger, wdsRestConfig *rest.Config, imbsRestConfig *rest.Config,
 	wdsName string, allowedGroupsSet sets.Set[string]) (*Controller, error) {
 
-	dynamicClient, err := dynamic.NewForConfig(wdsRestConfig)
+	kubernetesClient, err := kubernetes.NewForConfig(wdsRestConfig)
 	if err != nil {
 		return nil, err
 	}
 
-	kubernetesClient, err := kubernetes.NewForConfig(wdsRestConfig)
+	extClient, err := apiextensionsclientset.NewForConfig(wdsRestConfig)
 	if err != nil {
 		return nil, err
 	}
 
-	extClient, err := apiextensionsclientset.NewForConfig(wdsRestConfig)
+	burst, err := countGVRsViaAggregatedDiscovery(rest.CopyConfig(wdsRestConfig)) // use # of GVRs as burst is tested to be working well
```

Comment:
We could pass `kubernetesClient` here instead of `wdsRestConfig`, right? Passing the client would save a bit of source code and runtime cost; most importantly, it would make the counting func more easily testable.

Reply:
Good point. Changed as suggested.